c7n/resources/iam.py

```diff
@@ -791,23 +794,38 @@ class SetPolicy(BaseAction):
 
     permissions = ('iam:AttachRolePolicy', 'iam:DetachRolePolicy',)
 
+    def validate(self):
+        if self.data.get('state') == 'attached' and self.data.get('arn') == "*":
+            raise PolicyValidationError(
+                'State should be detached when arn is * on %s' % (self.manager.data))
```

Comment:
This could be clearer, i.e. `"*" operator is not supported for state: attached`

Reply:
Changed, thanks!